src/main/java/bio/terra/janitor/db/JanitorDao.java

```diff
@@ -188,18 +201,38 @@ public boolean updateResourceState(
   }
 
   /**
-   * Modifies the {@link CleanupFlightState} of a single flight. Returns whether there was a flight
-   * with the flightId.
+   * Modifies the {@link CleanupFlightState} of a single flight. Returns the updated CleanupFlight,
+   * if one was updated.
    */
   @Transactional(propagation = Propagation.SUPPORTS)
-  public boolean updateFlightState(String flightId, CleanupFlightState flightState) {
+  public Optional<CleanupFlight> updateFlightState(
+      String flightId, CleanupFlightState flightState) {
     String sql =
-        "UPDATE cleanup_flight SET flight_state = :flight_state WHERE flight_id = :flight_id;";
+        "UPDATE cleanup_flight SET flight_state = :flight_state WHERE flight_id = :flight_id "
+            + "RETURNING flight_id, flight_state";
     MapSqlParameterSource params =
         new MapSqlParameterSource()
             .addValue("flight_state", flightState.toString())
             .addValue("flight_id", flightId);
-    return jdbcTemplate.update(sql, params) == 1;
+    return Optional.ofNullable(
+        DataAccessUtils.singleResult(jdbcTemplate.query(sql, params, CLEANUP_FLIGHT_ROW_MAPPER)));
+  }
+
+  @Transactional(propagation = Propagation.SUPPORTS)
+  public Map<String, String> retrieveLabels(TrackedResourceId trackedResourceId) {
+    String sql =
+        "SELECT key, value FROM label " + "WHERE tracked_resource_id = :tracked_resource_id";
+    MapSqlParameterSource params =
+        new MapSqlParameterSource().addValue("tracked_resource_id", trackedResourceId.uuid());
+    List<AbstractMap.SimpleEntry<String, String>> labels =
+        jdbcTemplate.query(
+            sql,
+            params,
+            (rs, rowNum) ->
+                new AbstractMap.SimpleEntry<>(rs.getString("key"), rs.getString("value")));
+    return labels.stream()
+        .collect(
+            Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
```

Comment:
The tracked UID + key is the PK. There is no validations to check the duplicates of a key(not sure if we will have duplicate keys for a resource). If we want this way, we may need to add some checks during resource creation?

Reply:
That's already enforced by the JanitorDao API. Since we only can create Labels using a Map<String, String> with the tracked resource id, there's not a way to violate that. The sql insertion would also fail wby violating the foreign key constraint, so we don't need to check that postgres is maintaining its variants on retrieval.